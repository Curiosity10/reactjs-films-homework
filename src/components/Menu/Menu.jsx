import React from 'react';
import styles from './Menu.scss';
import FilterPanel from './components/FilterPanel';
import ChangeLayoutBtns from './components/ChangeLayoutBtns';

const Menu = () => (
  <section className={styles.menu}>
    <FilterPanel />
    <ChangeLayoutBtns />
  </section>
);

export default Menu;
