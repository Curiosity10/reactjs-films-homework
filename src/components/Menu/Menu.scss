@import "../../styles/variables.scss";

.menu {
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid #b7b7b7;
  padding-top: 40px;
  margin-bottom: 30px;
  margin-left: 5%;
  margin-right: 5%;

  & > nav {
    flex-grow: 1;
  }
}

@media all and (max-width: $tablet-screen-size) {
  .menu {
    padding-top: 25px;
    border-bottom: none;
    margin-bottom: 10px;
  }

  .menu li {
    margin-bottom: 15px;
  }
}
