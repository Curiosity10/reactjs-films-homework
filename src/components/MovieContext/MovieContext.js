import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { getPopularMovies } from '../../utils/tbdbApiService';

export const MovieContext = React.createContext();

export const MovieContextProvider = ({ children }) => {
  const [layout, setLayout] = useState('grid');
  const initialState = {
    movies: getPopularMovies(),
    layout,
    changeLayout: setLayout,
  };
  return (
    <MovieContext.Provider value={initialState}>
      {children}
    </MovieContext.Provider>
  );
};

MovieContextProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
