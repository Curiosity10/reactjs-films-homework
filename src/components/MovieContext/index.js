import { MovieContext, MovieContextProvider } from './MovieContext';

export { MovieContext, MovieContextProvider };
