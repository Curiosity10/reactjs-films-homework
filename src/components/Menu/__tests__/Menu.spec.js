import React from 'react';
import { create } from 'react-test-renderer';
import Menu from '..';
import { MovieContextProvider } from '../../MovieContext';

describe('Menu component renders correctly', () => {
  it('Menu renders correctly', () => {
    const tree = create(
      <MovieContextProvider>
        <Menu />
      </MovieContextProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
