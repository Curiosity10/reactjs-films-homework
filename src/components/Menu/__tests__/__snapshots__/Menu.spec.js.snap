// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders correctly Menu renders correctly 1`] = `
<section
  className="menu"
>
  <nav>
    <ul>
      <li
        className="navItem"
      >
        <button
          aria-label="Trending"
          className="navItem_btn"
          onClick={[Function]}
          type="button"
        >
          Trending
        </button>
      </li>
      <li
        className="navItem"
      >
        <button
          aria-label="Top Rated"
          className="navItem_btn"
          onClick={[Function]}
          type="button"
        >
          Top Rated
        </button>
      </li>
      <li
        className="navItem"
      >
        <button
          aria-label="Coming Soon"
          className="navItem_btn"
          onClick={[Function]}
          type="button"
        >
          Coming Soon
        </button>
      </li>
      <li
        className="navItem"
      >
        <select
          aria-label="Genres"
          className="navItem_selectGenre"
          name="selectGenre"
          onClick={[Function]}
        >
          <option
            disabled={true}
            hidden={true}
            selected={true}
          >
            Genre
          </option>
          <option
            value="Action"
          >
            Action
          </option>
          <option
            value="Adventure"
          >
            Adventure
          </option>
          <option
            value="Animation"
          >
            Animation
          </option>
          <option
            value="Comedy"
          >
            Comedy
          </option>
          <option
            value="Crime"
          >
            Crime
          </option>
          <option
            value="Documentary"
          >
            Documentary
          </option>
          <option
            value="Drama"
          >
            Drama
          </option>
          <option
            value="Family"
          >
            Family
          </option>
          <option
            value="Fantasy"
          >
            Fantasy
          </option>
          <option
            value="History"
          >
            History
          </option>
          <option
            value="Horror"
          >
            Horror
          </option>
          <option
            value="Music"
          >
            Music
          </option>
          <option
            value="Mystery"
          >
            Mystery
          </option>
          <option
            value="Romance"
          >
            Romance
          </option>
          <option
            value="Science Fiction"
          >
            Science Fiction
          </option>
          <option
            value="TV Movie"
          >
            TV Movie
          </option>
          <option
            value="Thriller"
          >
            Thriller
          </option>
          <option
            value="War"
          >
            War
          </option>
          <option
            value="Western"
          >
            Western
          </option>
        </select>
      </li>
    </ul>
  </nav>
  <div>
    <button
      aria-label="Grid"
      className="btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="#b7b7b7"
        height="20"
        viewBox="0 0 271.673 271.673"
        width="20"
        x="0px"
        y="0px"
      >
        <g>
          <g>
            <g>
              <path
                d="M114.939,0H10.449C4.678,0,0,4.678,0,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449V10.449C125.388,4.678,120.71,0,114.939,0z"
              />
              <path
                d="M261.224,0h-104.49c-5.771,0-10.449,4.678-10.449,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449V10.449C271.673,4.678,266.995,0,261.224,0z"
              />
              <path
                d="M114.939,146.286H10.449C4.678,146.286,0,150.964,0,156.735v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449v-104.49C125.388,150.964,120.71,146.286,114.939,146.286z"
              />
              <path
                d="M261.224,146.286h-104.49c-5.771,0-10.449,4.678-10.449,10.449v104.49c0,5.771,4.678,10.449,10.449,10.449h104.49 c5.771,0,10.449-4.678,10.449-10.449v-104.49C271.673,150.964,266.995,146.286,261.224,146.286z"
              />
            </g>
          </g>
        </g>
      </svg>
    </button>
    <button
      aria-label="Table"
      className="btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="#b7b7b7"
        height="20"
        viewBox="0 0 512 512"
        width="20"
        x="0px"
        y="0px"
      >
        <g>
          <g>
            <g>
              <rect
                height="235"
                width="100%"
              />
              <rect
                height="235"
                width="100%"
                y="277.333"
              />
            </g>
          </g>
        </g>
      </svg>
    </button>
  </div>
</section>
`;
